### ISSUE TITLE
Disable inotify

### ISSUE THREAD

**Issue by Porco-Rosso** [ejurgensen/forked-daapd#554]
Hi, Is there a way to disable Inotify? Although I increased the inotify max value I still exhaust all the memory. I get the following message, and believe it may be causing some other problems (maybe the eventual failure of websockets? https://github.com/chme/forked-daapd-web/issues/3 )

```
scan: Could not create inotify watch for /mnt/music/Album/songs: No space left on device
```

**Comment by ejurgensen** [comment]
No, you would have to comment out these lines https://github.com/ejurgensen/forked-daapd/blob/master/src/library/filescanner.c#L831 and then build yourself.

Is this Linux? How many directories do you have in your library? I haven't heard about anyone exhausting inotify before, but I suppose it could happen with very large libraries.

**Comment by Porco-Rosso** [comment]
I'm running debian on a VM (openvz container), with the music directory bind-mounted. I suspect it has something to do with the way the filesystem is mounted. 
My library has around ~10,000 tracks, but I believe another process on the same container is also using inotify on the library, so maybe double of that.

I found forked-daapd using 100% of the cpu this morning after using it yesterday, so something is definitely off. I'm planning to migrate to lxc containers, today so I will see if that makes a change. If not I will try commenting out the lines as you suggested.

I would suggest adding a build argument to disable inotify -- who knows, maybe someone will have a library that is really colossal.

**Comment by ejurgensen** [comment]
That size library shouldn't be a problem, so perhaps the root cause is elsewhere.

If you see 100% cpu then check the log to see what forked-daapd is doing. I think you can also what thread is busy using something like htop.

**Comment by Porco-Rosso** [comment]
So after reinstalling the OS in an lxc container, and recompiling forked-daapd, the 100% cpu usage problems have gone away. I still had some issues with inotify hitting it's resource limit, but at the end I figured it out. The container shared the inotify constraint with the host, so I needed to [increase the max value](https://unix.stackexchange.com/questions/13751/kernel-inotify-watch-limit-reached) there as well.
I feel this issue is resolved and at the end was not a problem so much with forked-daapd, as it was my implementation. Thanks for helping troubleshoot, and letting me know how to disable inotify if needed!
